core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -243,13 +246,8 @@ public void resetState() {
     mBlockMaster.registerWorkerLostListener(mWorkerConfigStore::handleNodeLost);
     mBlockMaster.registerNewWorkerConfListener(mWorkerConfigStore::registerNewConf);
 
-    if (URIUtils.isLocalFilesystem(ServerConfiguration
-        .get(PropertyKey.MASTER_MOUNT_TABLE_ROOT_UFS))) {
-      mUfs = UnderFileSystem.Factory
-          .create("/", UnderFileSystemConfiguration.defaults(ServerConfiguration.global()));
-    } else {
-      mUfs = UnderFileSystem.Factory.createForRoot(ServerConfiguration.global());
-    }
+    mRootUfsClient = masterContext.getUfsManager().getRoot();
+    mRootUfs = mRootUfsClient.acquireUfsResource();
```

Comment:
Does it make sense to keep these as variables, or could we acquire them when necessary?

Reply:
changed to acquire the resource when necessary. I think the code becomes more consistent and easier to reason.